src/ckeditor.js

```diff
@@ -12,9 +14,56 @@
  * @singleton
  */
 
-CKEDITOR.define( function() {
+CKEDITOR.define( [ 'editor', 'mvc/collection', 'promise' ], function( Editor, Collection, Promise ) {
 	var CKEDITOR = {
 		/**
+		 * A collection containing all editor instances created.
+		 */
+		instances: new Collection(),
+
+		/**
+		 * Creates an editor instance for the provided DOM element.
+		 *
+		 * The creation of editor instances is an asynchronous operation, therefore a promise is returned by this
+		 * method.
+		 *
+		 *     CKEDITOR.create( '#content' );
+		 *
+		 *     CKEDITOR.create( '#content' ).then( function( editor ) {
+		 *         // Manipulate "editor" here.
+		 *     } );
+		 *
+		 * @param {String|Object} element An element selector or a DOM element, which will be the source for the
+		 * created instance.
+		 * @returns {Promise} A promise, which will be fulfilled with the created editor.
+		 */
+		create: function( element ) {
+			var that = this;
+
+			return new Promise( function( resolve, reject ) {
+				// If a query selector has been passed, transform it into a real element.
+				if ( typeof element == 'string' ) {
+					element = document.querySelector( element );
```

Comment:
I remember that the original concept included this feature but it was not included in [the docs](https://github.com/ckeditor/ckeditor5-design/wiki/Editor-Instances) and it therefore it wasn't implemented.

But in such case, how to we want to resolve the promise? Right now it passes the (one) created editor instance. **If** we decide to make this change, we have the follow options:

 1. Pass an array back in case of multiple editors and an Editor instance in case of single. Error if none.
 2. Always pass an array back, even for one editor only. Empty array if no element found.
 3. Always pass an array back. Error if no element found.

For consistency, I would go with "3", but it'll be a pain in the ass for developers... especially because those that will use `then()` will usually be creating a single editor only.

Therefore, I think we may consider not having this feature right now.